File: 백준/Bronze/25501. 재귀의 귀재/재귀의 귀재.js
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').toString().trim().split('\n');
const T = +input[0];
let answer = '';

for (let i = 1; i <= T; i++) {
    const str = input[i];
    let count = 0;
    
    function recursion(left, right) {
        count++;
        
        if (left >= right) return 1;
        else if (str[left] !== str[right]) return 0;
        else return recursion(left + 1, right - 1);
    }
    
    answer += `${recursion(0, str.length - 1)} ${count}\n`;
}

console.log(answer);

File: 백준/Bronze/25501. 재귀의 귀재/재귀의 귀재.py
import sys

T = int(input())

def recursion(count, S, l, r):
    if l >= r:
        return (1, count+1);
    elif S[l] != S[r]:
        return (0, count+1)
    else:
        return recursion(count+1, S, l+1, r-1)

for S in sys.stdin.read().splitlines():
    print(*recursion(0, S, 0, len(S)-1))